subsys/bluetooth/audio/csis_client.c

```diff
@@ -140,6 +143,110 @@ static struct bt_csis *lookup_instance_by_set_info(const struct bt_csis_client_s
 	return NULL;
 }
 
+static int member_rank_compare_asc(const void *m1, const void *m2)
+{
+	struct bt_csis_client_set_member *member_1 = *(struct bt_csis_client_set_member **)m1;
+	struct bt_csis_client_set_member *member_2 = *(struct bt_csis_client_set_member **)m2;
+	struct bt_csis *inst_1;
+	struct bt_csis *inst_2;
+
+	/* TODO: lookup_instance_by_set_info should be able to work on a const
+	 * member, but it relies on `bt_conn_index` which doesn't take a const
+	 * bt_conn.
+	 */
+
+	inst_1 = lookup_instance_by_set_info(member_1, active.info);
+	inst_2 = lookup_instance_by_set_info(member_2, active.info);
+
+	if (inst_1 == NULL) {
+		BT_ERR("inst_1 was NULL for member %p", member_1);
+		return 0;
+	}
+
+	if (inst_2 == NULL) {
+		BT_ERR("inst_2 was NULL for member %p", member_2);
+		return 0;
+	}
+
+	if (inst_1->cli.rank < inst_2->cli.rank) {
+		return -1;
+	} else if (inst_1->cli.rank > inst_2->cli.rank) {
+		return 1;
+	}
+
+	return 0;
+}
+
+static int member_rank_compare_desc(const void *m1, const void *m2)
+{
+	struct bt_csis_client_set_member *member_1 = *(struct bt_csis_client_set_member **)m1;
+	struct bt_csis_client_set_member *member_2 = *(struct bt_csis_client_set_member **)m2;
+	struct bt_csis *inst_1;
+	struct bt_csis *inst_2;
+
+	/* TODO: lookup_instance_by_set_info should be able to work on a const
+	 * member, but it relies on `bt_conn_index` which doesn't take a const
+	 * bt_conn.
+	 */
+
+	inst_1 = lookup_instance_by_set_info(member_1, active.info);
+	inst_2 = lookup_instance_by_set_info(member_2, active.info);
+
+	if (inst_1 == NULL) {
+		BT_ERR("inst_1 was NULL for member %p", member_1);
+		return 0;
+	}
+
+	if (inst_2 == NULL) {
+		BT_ERR("inst_2 was NULL for member %p", member_2);
+		return 0;
+	}
+
+	if (inst_1->cli.rank < inst_2->cli.rank) {
+		return 1;
+	} else if (inst_1->cli.rank > inst_2->cli.rank) {
+		return -1;
+	}
+
+	return 0;
+}
```

Comment:
Are those separate functions needed just to return -1 vs 1? Please extract common part

Reply:
The only thing I can see that I can make common is moving the casting of `const void *` to `struct bt_csis_client_set_member *`, which would require me to create a function that takes a `const void *` that then casts this to a `struct bt_csis_client_set_member *` and then does the call to `lookup_instance_by_set_info`. 

But that extra function would just remove two lines 
```
	struct bt_csis_client_set_member *member_1 = *(struct bt_csis_client_set_member **)m1;
	struct bt_csis_client_set_member *member_2 = *(struct bt_csis_client_set_member **)m2;
```
and replace 
```c
	inst_1 = lookup_instance_by_set_info(member_1, active.info);
```
with something like
```c
	inst_1 = member_rank_compare_lookup_instance_by_set_info(m1, active.info);
```

I wanted to use `qsort_r` which allows for an additional argument, which could also make this a single function instead, but that is not POSIX compliant, and won't work with e.g. babblesim nor native_posix. 
The alternative is to store the comparison direction (ascending/descending) in the `active` struct. 

The last alternative is to create a new struct that holds a tuple of `struct bt_csis *` and return that. That way it could made common. 

If you want, I can create the tuple struct return type, and extract common code, or I can add a sorting direction variable in `active`, which would reduce this to a single function. Which do you prefer?

Reply:
Try this one:
```
static int member_rank_compare_desc(const void *m1, const void *m2)
{
    return member_rank_compare_asc(m2, m1);
}
```

Reply:
That's a great suggestion - Did not think of that myself :D